Wrappers/Python/test/test_out_in_place.py

```diff
@@ -313,3 +315,155 @@ def test_proximal_out(self):
+
+        ag_parallel_2D = AcquisitionGeometry.create_Parallel2D()
+        angles = np.linspace(0, 360, 10, dtype=np.float32)
+        ag_parallel_2D.set_angles(angles)
+        ag_parallel_2D.set_panel(20)
+
+        ag_parallel_3D = AcquisitionGeometry.create_Parallel3D()
+        ag_parallel_3D.set_angles(angles)
+        ag_parallel_3D.set_panel([20,2])
+
+        ag_cone_2D = AcquisitionGeometry.create_Cone2D(source_position=[0,-10],detector_position=[0,10])
+        ag_cone_2D.set_angles(angles)
+        ag_cone_2D.set_panel(20)
+
+        ag_cone_3D = AcquisitionGeometry.create_Cone3D(source_position=[0,-10,0],detector_position=[0,10,0])
+        ag_cone_3D.set_angles(angles)
+        ag_cone_3D.set_panel([20,2])
+        
+        self.geometry_test_list = [ag_parallel_2D, ag_parallel_3D, ag_cone_2D, ag_cone_3D]
+        self.data_test_list= [geom.allocate(None) for geom in self.geometry_test_list]
+        
+        flat_field = self.data_test_list[0]*1
+        dark_field = self.data_test_list[0]*1e-5
+
+        self.processor_list = [
+            TransmissionAbsorptionConverter(min_intensity=0.01),
+            AbsorptionTransmissionConverter(),
+            RingRemover(),
+            Slicer(roi={'horizontal':(None,None,2),'angle':(None,None,2)}), 
+            Binner(roi={'horizontal':(None,None,2),'angle':(None,None,2)}), 
+            Padder(pad_width=1)]
+        self.processor_out_same_size = [
+            True,
+            True,
+            True,
+            False,
+            False,
+            False
+        ]
+
+    def get_result(self, processor, data, *args):
+        input=data.copy() #To check that it isn't changed after function calls
+        try:
+            processor.set_input(data)
+            out = processor.get_output()
+            self.assertDataArraysInContainerAllClose(input, data, rtol=1e-5, msg= "In case processor.set_input(data), processor.get_output() where processor is  " + processor.__class__.__name__+ " the input data has been incorrectly affected by the calculation.")
+            return out
+        except NotImplementedError:
+            print("get_result test not implemented for  " + processor.__class__.__name__)
+            return None
+
+    def in_place_test(self, desired_result, processor, data, output_same_size=True):
+            if output_same_size==True:
+                out = data.copy()
+            else:
+                out=desired_result.copy()
+            try:
+                processor.set_input(data)
+                processor.get_output(out=out)
+                self.assertDataArraysInContainerAllClose(desired_result, out, rtol=1e-5, msg= "In place calculation failed for processor.set_input(data), processor.get_output(out=data) where processor is  " + processor.__class__.__name__+ "." )
+
+            except (InPlaceError, NotImplementedError):
+                print("in_place_test test not implemented for  " + processor.__class__.__name__)
+                pass
```

Comment:
I think the focus of this test, should be if the following works:
```python
processor.set_input(data)
processor.get_output(out=data)
self.assertDataArraysInContainerAllClose(desired_result, data)
```

This is only relevant if the sizes of input and output remain the same.  At the moment this test is the same as the `out_test`

Reply:
Good point, I've updated this and now only using it for same size input and output